update: adding compatibility functions for rtl languages.

moakilodash opened the pull request:
- main-layout
 - flying menu
 - mobile menu
 - ...

Commit:
update: adding compatibility functions for rtl languages. - main-layout - flying menu - mobile menu - ...

kevinmulier commented on apps/web/src/components/Header/FlyingMenu/flying-menu-section.tsx:
```diff
@@ -148,6 +171,7 @@ export const FlyingMenuSection = ({ section, variant }: FlyingMenuProps) => {
           className={cn(
             'flex-auto overflow-hidden rounded-[20px]',
             variant === 'light' ? 'bg-darkOrange-2' : 'bg-newBlack-3',
+            rtl ? 'text-red-5' : '',
```
Can your remove this please ?

kevinmulier commented on apps/web/src/components/Header/FlyingMenu/flying-menu-sub-section.tsx:
```diff
@@ -13,14 +15,17 @@ export const FlyingMenuSubSection = ({
             'text-lg font-primary font-semibold',
             'items' in subSection ? 'mb-4 px-2' : '',
             variant === 'light' ? 'text-black' : 'text-white',
+            rtl ? 'text-red-5' : '',
```
This can be removed as well

kevinmulier commented on apps/web/src/components/Header/MobileMenu/mobile-menu.tsx:
```diff
@@ -156,8 +158,15 @@ export const MobileMenu = ({
+          'flex flex-col fixed top-0 items-center w-[90%] max-w-[440px] h-dvh pb-5 duration-300 overflow-scroll no-scrollbar lg:hidden bg-darkOrange-2 dark:bg-newBlack-2',
+          rtl
+            ? 'right-0 border-l border-darkOrange-4 dark:border-l dark:border-newBlack-4'
+            : 'left-0 border-r border-darkOrange-4 dark:border-r dark:border-newBlack-4',
```
duplicated parts should go in the main classes string

kevinmulier commented on apps/web/src/components/Header/MobileMenu/mobile-menu.tsx:
```diff
@@ -174,12 +183,21 @@ export const MobileMenu = ({
             size={24}
-            className="text-maroon-7 dark:text-newGray-3 shrink-0 ml-auto cursor-pointer"
+            className={cn(
+              'text-newGray-3 shrink-0 cursor-pointer',
```
should keep text-maroon-7 / dark:text-newGray-3

kevinmulier commented on apps/web/src/components/Header/FlyingMenu/flying-menu.tsx:
```diff
@@ -34,15 +42,16 @@ export const FlyingMenu = ({
           variant === 'light'
             ? 'bg-darkOrange-2 text-black'
             : 'bg-newBlack-3 text-white',
+          rtl ? 'text-red-5' : '',
```
This text-red-5 should be removed

kevinmulier commented on apps/web/src/components/Header/FlyingMenu/flying-menu.tsx:
```diff
@@ -34,15 +42,16 @@ export const FlyingMenu = ({
       <ul
         className={cn(
-          'mx-auto flex flex-row items-center gap-2 xl:gap-5 rounded-xl px-3 py-2.5',
+          'flex flex-row items-center gap-2 xl:gap-5 rounded-xl px-3 py-2.5',
```
Why is mx-auto removed here ?

moakilodash replied:
such a good point, thanks! i wanted to actually fix the logo alignment on rtl.
probably cursor tab removed it by accident.

kevinmulier commented on apps/web/src/components/Header/FlyingMenu/flying-menu-section.tsx:
```diff
@@ -199,6 +223,7 @@ export const FlyingMenuSection = ({ section, variant }: FlyingMenuProps) => {
                     <MenuElement
                       element={subSectionOrElement}
                       variant={variant}
+                      isMultipleSubSectionChildren={hasMultipleSubSection}
```
Why was this added ?

moakilodash replied:
yeah since there is no need for width control logic for the "sub-elements" it has no purpose, i removed it.

kevinmulier commented on apps/web/src/components/Header/FlyingMenu/flying-menu-section.tsx:
```diff
@@ -108,9 +128,10 @@ export const FlyingMenuSection = ({ section, variant }: FlyingMenuProps) => {
         className={cn(
           'relative px-2 xl:px-4 py-1.5 rounded-lg hover:bg-white/20',
           open && 'bg-white/20',
+          rtl && 'text-red-5',
```
Can you remove this text-red-5 ?

kevinmulier reviewed (changes requested):
Hey, thanks a lot for taking the time to fix our RTL layout, it was much needed!
Could you also take a look at the few points I left in my review if you have some time left?

Thanks again 😊

Commit:
fix: moved duplicated classes to parent, and removed troubling parts from the review on pr#1619

moakilodash commented:
sorry for late response...
i modified the parts you asked in your review.
but i have to mention that there are still incompatible rtl parts in the layout, for example:
- the text alignment and text shortening in the flying-menu-subsection that has to start from right.
- icons that has to be on the right side of the text where, and icons that has to change direction. (for example from `Check all events >` to `< بررسی تمام رویداد‌ها`)
- "Plan ₿" that has been written "₿ Plan" only in some texts in rtl.
- and possible design inconsistencies throughout the app...

kevinmulier commented on apps/web/src/components/Header/FlyingMenu/flying-menu-section.tsx:
```diff
@@ -124,6 +144,7 @@ export const FlyingMenuSection = ({ section, variant }: FlyingMenuProps) => {
           className={cn(
             'relative px-2 xl:px-4 py-1.5 rounded-lg hover:bg-white/20',
             open && 'bg-white/20',
+            rtl && 'text-red-5',
```
Hey can you remove this as well ?

kevinmulier commented on apps/web/src/components/Header/FlyingMenu/flying-menu-sub-section.tsx:
```diff
@@ -13,11 +15,13 @@ export const FlyingMenuSubSection = ({
+  const rtl = isRTL(i18n.language);
   return (
-    <div className="flex flex-col my-2 mx-2.5 gap-2.5">
+    <div className={cn('flex flex-col my-2 mx-2.5 gap-2.5')}>
```
we can remove cn here since it's only a single string

kevinmulier commented on apps/web/src/components/Header/FlyingMenu/flying-menu-sub-section.tsx:
```diff
@@ -13,11 +15,13 @@ export const FlyingMenuSubSection = ({
   hasMultipleSubSection,
 }: FlyingMenuSubSectionProps) => {
+  const { i18n } = useTranslation();
+  const rtl = isRTL(i18n.language);
```
this is unused here, we can safely remove the rtl and i18n const (remove it in the imports as well please)

kevinmulier commented on apps/web/src/components/Header/FlyingMenu/flying-menu.tsx:
```diff
@@ -23,9 +24,16 @@ export const FlyingMenu = ({
+        rtl && 'flex-row-reverse text-red-5',
+      )}
+    >
+      <Link to="/" className={rtl ? 'ml-auto' : 'mr-auto'}>
```
this is creating some issue with the layout (logo should be close to the edge, it is not anymore with that change)

revert to "mr-auto" for both cases

moakilodash replied:
alright, i also deleted reversing the rows as the logo is placed on the right side of the screen in rtl websites, i.e. [a farsi online shop](https://www.digikala.com/)

kevinmulier commented on apps/web/src/components/Header/FlyingMenu/flying-menu.tsx:
```diff
@@ -23,9 +24,16 @@ export const FlyingMenu = ({
+    <nav
+      className={cn(
+        'flex w-full flex-row items-center justify-between max-lg:hidden',
+        rtl && 'flex-row-reverse text-red-5',
```
the text-red should be removed here

kevinmulier reviewed (commented):
Hello ! Thank you for making the changes and giving us hints about the parts that are still not RTL-friendly

As we are making a lot of changes on the UI/UX (especially on menu), we will have to see how the app behaves once the changes are implemented

In the meantime (as the big UI/UX changes are gonna be released in 1-2 months, not immediately) we will implement your changes so that the experience of users with RTL language will be improved

I added a few comments/reviews, if you can spend some time on them it would be great ! Thanks !

---

<img width="1683" height="343" alt="Image" src="https://github.com/user-attachments/assets/1bd69248-794f-4e15-9122-0b6ef77a7041" />

-> And if you can fix that as well it would be good, maybe get the sub menu to open on the left when in RTL so there is enough space to display them entirely ?

Thank you in advance for your work !

Commit:
chore: updating the rtl functionality of flying menu elements and language selector components, and adding vazirmatn font for farsi language

moakilodash commented:
hey @kevinmulier,
i spent a little time on trying to fix some of the points i mentioned in my previous comment,
now the flying menu works correctly with one caveat that the ellipsis for menu-elements shows in the beginning of a the text, which i couldn't figure why.
i added a more easier to the eye font for farsi language, since i can read the language and i figured this wouldn't be that opinionated to choose a font, and basically anything would be better than browser's default font for this language.

i think i covered everything you had in your reviews by now, so let me know if there is anything else you want me to change in my recent commit.

one more thing to mention is that i've added many flex direction changes to these components that may seem redundant, but there must be a decision in terms of what should/shouldn't be rtl in rtl languages. either for keeping a coherent ltr for main components, or different designs and thinking process behind each of them...

moakilodash commented:
also regarding the '>' and '<' icons, in my opinion it's better to have different components for each that gets added depending if `rtl` is enabled or not.
let me know what you think about this.

kevinmulier commented:
Hey @moakilodash , thanks a lot for the work you’ve put into this PR and for taking the time to improve the RTL (and Farsi) experience, it will be really helpful for our users

Since we’re in the middle of rolling out a new UI and the timing is a bit sensitive, I’ll merge it and make some small adjustments to your changes internally (mainly around the font and some layout details) so we can merge everything without delaying the release
I just wanted to let you know so it doesn’t feel like I’m overriding your work, it’s really about keeping things aligned with the new UI / internal design changes

Once this new UI is stable (in around a month), I’d love for you to keep helping us polish the RTL experience if you wish to do so and see any issue that could be fixed on your side

Thanks again
